pageserver/src/tenant/blob_io.rs

```diff
@@ -66,12 +69,29 @@ impl<'a> BlockCursor<'a> {
                 len_buf.copy_from_slice(&buf[off..off + 4]);
                 off += 4;
             }
-            len_buf[0] &= 0x7f;
+            len_buf[0] &= 0x0f;
             u32::from_be_bytes(len_buf) as usize
         };
+        let compression_bits = first_len_byte & 0xf0;
```

Comment:
Please avoid inline hex & use named constants for bitmasks, like "LEN_HEADER_COMPRESSION_MASK" (then where you use 0x0f above you can invert the mask, and it becomes more obvious that than lines 72 & 75 are just two sides of the same coin.

Reply:
done